src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -363,30 +364,37 @@ class QueryCompiler(cq : ConjunctiveQuery, ss: CompilationState) {
         resolved
       }
       case BinaryOpExpr(lhs, op, rhs) => {
-        val resovledLhs = compileExpr(lhs)
-        val resovledRhs = compileExpr(rhs)
-        s"(${resovledLhs} ${op} ${resovledRhs})"
+        val resovledLhs = compileExpr(lhs, level + 1)
+        val resovledRhs = compileExpr(rhs, level + 1)
+        val sql = s"${resovledLhs} ${op} ${resovledRhs}"
+        if (level == 0) sql else s"(${sql})"
```

Comment:
That's exactly true.  I think compiling a more compact output is generally a good thing, no?  Otherwise, the parentheses should have been placed around every expression rather than just the case for binary operations to make the compiler code itself more succinct.